### PR TITLE
fix(admin): id in NodeInfo is string instead of B256

### PR DESCRIPTION
**Pull request by jsvisa**
## Motivation

Return the same response type as geth's, ref https://github.com/ethereum/go-ethereum/blob/v1.14.0/p2p/server.go#L1078-L1079

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/node-bindings/src/geth.rs [hunk 1]
```diff
@@ -121,7 +121,7 @@ impl GethInstance {
     /// Blocks until geth adds the specified peer, using 20s as the timeout.
     ///
     /// Requires the stderr to be present in the `GethInstance`.
-    pub fn wait_to_add_peer(&mut self, id: B256) -> Result<(), GethInstanceError> {
+    pub fn wait_to_add_peer(&mut self, id: String) -> Result<(), GethInstanceError> {
```
Comment (DaniPopes) [inline crates/node-bindings/src/geth.rs:124]: use &str

File: crates/node-bindings/src/geth.rs [hunk 2]
```diff
@@ -132,8 +132,8 @@ impl GethInstance {
             err_reader.read_line(&mut line).map_err(GethInstanceError::ReadLineError)?;
 
             // geth ids are truncated
-            let truncated_id = hex::encode(&id.0[..8]);
-            if line.contains("Adding p2p peer") && line.contains(&truncated_id) {
+            let truncated_id = &id[0..8];
```
Comment (DaniPopes) [inline crates/node-bindings/src/geth.rs:135]: this panics if too short

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by mattsse — COMMENTED** [review]
nice find, I believe I couldn't figure out what the correct value here is

File: crates/node-bindings/src/geth.rs (hunk 2 above)
Comment (jsvisa) [reply crates/node-bindings/src/geth.rs]: thanks for the tip, then I change it to get the first 16bytes instead.

**Review by mattsse — APPROVED** [review]
lgtm
ty

@DaniPopes if we release alloy with this fix alloy as a patch, this will break 1.0.1 reth build for some folks

**Pull request merged**